Source/Core/Resource.js

```diff
@@ -653,9 +653,15 @@ Resource.prototype.getDerivedResource = function (options) {
     parseQuery(uri, resource, true, preserveQueryParameters);
 
     // Remove the fragment as it's not sent with a request
-    uri.fragment = undefined;
+    uri.fragment("");
 
-    resource._url = uri.resolve(new Uri(getAbsoluteUri(this._url))).toString();
+    if (uri.is("urn")) {
```

Comment:
`URI.absoluteTo()` throws an exception when trying to concatenate URNs, whereas Uri.js returned the unchanged URN. Is this the correct behavior for `Resource`? There are a couple other places where this came up.

Reply:
Where exactly are the `urn` urls coming from?

Reply:
It's possible in .gltf files, i.e. the `images` in https://github.com/CesiumGS/cesium/blob/main/Specs/Data/Models/GltfLoader/BoxTextured/glTF-Embedded/BoxTextured.gltf, used in -https://github.com/CesiumGS/cesium/blob/1837859300e61ae2ea1bc0e3832923f87e752fed/Specs/Scene/GltfLoaderSpec.js#L48.

Not sure how common this is..

Reply:
I thought data uris were different than urn?

According to https://stackoverflow.com/a/28865747 urn's start with `urn:`

Reply:
Yeah, there's definitely confusion here.  almost everything is an urn.. so I'm not sure what differentiates from data uri or other in `is`

Reply:
Ah that's my mistake.. I misunderstood the terminology. But, now I realize there's a cleaner fix - checking whether the url's `scheme === "data"` should do the trick and is more readable I think.

Reply:
Yeah, that's a check we do in a bunch of places (we even have an `isDataUri` helper function for it)

Reply:
On second thought, I don't think just checking for data uris is enough - `ModelOutlineLoaderSpec` catches cases in Resource.getDerivedResource where the scheme is `"blob"`. The third party Uri.js in `main` ignores the base URL's properties (including scheme) if the relative URL has them set: 

https://github.com/CesiumGS/cesium/blob/1837859300e61ae2ea1bc0e3832923f87e752fed/Source/ThirdParty/Uri.js#L177-L182

Sandcastles and specs pass if everywhere that (incorrectly) checks `Uri.is("urn")` is changed to `Uri.scheme() === ""`. This happens in `RequestScheduler.getServerKey`, `Resource.getDerivedResource`, and `getAbsoluteUri`. Is checking just for data schemes as opposed to checking for any scheme before calling `Uri.absoluteTo` preferable anywhere? Checking for any scheme at all is more faithful to the behavior in `main`.

Reply:
So what is the behavior for these urls in `main`?  Do we just do nothing and return the original url?  If so, then we can do the same thing here for now, but it could ultimately be masking a bug. Like should we be trying to absolutize or otherwise manipulate a opaque URL at all?

Reply:
And wouldn't `is("relative")` be what we would want to check instead?

Reply:
It's basically the same check under the hood, but much clearer that we are only doing certain things for relative paths

Reply:
> Do we just do nothing and return the original url?

Yes, this is what the current third party Uri.js does.

> And wouldn't is("relative") be what we would want to check instead?

Not quite, since doing so falsely flags urls like `"test.png"`, which causes spec failures. In main, `new Uri("test.png").resolve(new Uri("http://localhost:8080/Specs/SpecRunner.html"))` turns into `"http://localhost:8080/Specs/test.png"`. This is consistent with this npm module's `absoluteTo`.